consensus/src/threshold_simplex/actors/resolver/actor.rs

```diff
@@ -502,74 +513,119 @@ impl<
                             };
                             self.inflight.clear(request.id);
 
-                            // Parse notarizations
-                            let mut notarizations_found = BTreeSet::new();
-                            let mut nullifications_found = BTreeSet::new();
+                            // Compute batch
+                            let mut notarizations_found = BTreeMap::new();
+                            let mut nullifications_found = BTreeMap::new();
+                            let mut batch = HashMap::new();
+                            let mut invalid = false;
                             for notarization in response.notarizations {
                                 let view = notarization.view();
                                 let entry = Entry { task: Task::Notarization, view };
                                 if !self.required.contains(&entry) {
                                     debug!(view, sender = ?s, "unnecessary notarization");
                                     continue;
                                 }
-                                let Some(identity) = self.supervisor.identity(view) else {
-                                    warn!(view, sender = ?s, "missing identity");
-                                    continue;
-                                };
-                                let public_key = poly::public::<V>(identity);
-                                if !notarization.verify(&self.namespace, public_key) {
-                                    warn!(view, sender = ?s, "invalid notarization");
-                                    self.requester.block(s.clone());
-                                    continue;
+                                let items = notarization.message(&self.namespace);
+                                let mut unique = 0;
+                                for (namespace, message, signature) in items {
```

Comment:
Clear UX issue is around creating an aggregate signature from a batch of messages with potentially overlapping inner verification messages (seeds).